Add support for EL10

middelkoopt opened the pull request:
## Description of the Pull Request (PR):

Add support for EL10
 * Update `.spec` to depend on `dnsmasq` and not `dhcpd-server` for EL10
 * Add make rpm and version targets to the makefile
 * Use `tmpfile` for `make dist` to support building in a devcontainer with colima on a mac.
 * Update CI/CD to get the version via `make version` to support multiple directories in the mock build process (glob returns multiple directories in the `make -C` in `release.yml`.
 * Update CI/CD to use Rocky 10 build container.
 * Update CI/CD to add EL10 target
 * Update CI/CD to use `make version` information for mock
 * Updated .devcontainer to Rocky 10

Only tested on a manual trigger.

## Reviewer  checklist

The reviewer checks the following items before merging the PR.

- [x] The PR is based on the appropriate branch (typically [main](https://github.com/warewulf/warewulf/tree/main/userdocs))
- [x] All commits are "Signed off" (e.g., using `git commit --signoff`) in agreement to the [DCO](DCO.txt)
- [x] The [CHANGELOG](https://github.com/warewulf/warewulf/blob/main/CHANGELOG.md) has been updated, if necessary, and under the correct release heading
- [ ] The [userdocs](https://github.com/warewulf/warewulf/tree/main/userdocs) have been updated, if necessary
- [x] The submitter is listed in the [contributors file](https://github.com/warewulf/warewulf/blob/main/CONTRIBUTORS.md)
- [x] The test suite has been updated, if necessary

middelkoopt commented on Makefile:
```diff
@@ -28,14 +32,21 @@ spec: warewulf.spec ## Create an RPM spec file
+	tar -C $(TMPDIR) -czf $(WAREWULF)-$(VERSION).tar.gz $(WAREWULF)-$(VERSION)
+	rm -rf $(TMPDIR)/
+
+RPMDIR = $(HOME)/rpmbuild
+.PHONY: rpm
+rpm: spec dist ## Create an RPM package
+	@mkdir -p $(RPMDIR)/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+	cp $(WAREWULF)-$(VERSION).tar.gz $(RPMDIR)/SOURCES/
+	rpmbuild -bb warewulf.spec
```
I added it to be able to easily test the RPM during dev, and figured it would be useful for users to create an RPM and install it either for local use to aid in contributing code to Warewuf. I can remove it.

anderbubble replied:
Oh. If it's just there for convenience and not to use during the GH automation, that's fine.

middelkoopt commented:
I went ahead and finished out the EL10 support.  I also did a lot more testing this time round on  alma/rocky/8/9/10/leap/aarch64/x86_64 as well as OpenHPC 4.x EL10 builds (fixes not pushed) .

* Fixed the inverted issue for `dnsmasq` that you identified
* Removed dhcp-server and tftp-server from non EL10 builds
* Modified `warewulf.conf` to use `dnsmasq` by default
* Fix dnsmasq template tftp paths to use basename (this only impacted aarch64)

I feel good about the state the PR is in now.  Next up getting Warewulf and EL10 working for OpenHPC 4.x.

middelkoopt commented on warewulf.spec.in:
```diff
@@ -60,12 +60,16 @@ Requires: ipxe-bootimgs-aarch64
 %endif
 
-%if 0%{?rhel} >= 8 || 0%{?suse_version} || 0%{?fedora}
+%if 0%{?rhel} >= 10 || 0%{?suse_version} || 0%{?fedora}
```
This has been addressed and tested.

anderbubble commented:
@middelkoopt I think this is just blocking on me. It's on my list for today.
